Chore/sourcehooks remote fix

nvyasadobe opened the pull request:
Address issue: beforeSource/afterSource remote hook not working.

## Description

This PR address issue that beforeSource and afterSource hooks were not working with remoteJS function. Also added support in afterAll for extensions.

## Related Issue

https://jira.corp.adobe.com/browse/CEXT-1441

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

Currently tested against run command and making a local link of plugin-hooks dependency on local.

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

nvyasadobe committed:
chore: WIP: afterSource hook fix for remote hook

nvyasadobe committed:
feat: fixed the beforeSource remote hooks

nvyasadobe committed:
feat: minor correction and updated headers merge logic

nvyasadobe committed:
feat: added extensions support for afterAll

kmaschi commented on src/afterAllExecutor.ts:
```diff
@@ -77,6 +71,7 @@ export async function executeAfterAllHook(
 			setResultAndStopExecution({
 				data: hookResponse.data.result.data || result.data,
 				errors: hookResponse.data.result.errors || result.errors,
+				extensions: hookResponse.data.result.extensions || result.extensions,
```
This is a great use case for extensions. If a tenant wants to include custom fields it should be done in the extensions map like this. We should make sure documentation includes an example usecase.

Commit:
chore(remote-fix) - Removed unnecessary executors since invocated differently.
- Updated types to make clear the properties available to each hook handler.
- Handled stacked beforeAll/afterAll hooks ensuring both local and remove functions can work with serialized payload.
- Handled stacked beforeAll/afterAll hooks ensuring handlers can work with serialized response data.
- Updated tests.
